dj_rql/drf/backend.py

```diff
@@ -25,15 +25,38 @@ class ViewSet(mixins.ListModelMixin, GenericViewSet):
     """
     OPENAPI_RETRIEVE_SPECIFICATION = False
 
+    _CACHES = {}
+
     def filter_queryset(self, request, queryset, view):
         filter_class = self.get_filter_class(view)
         if not filter_class:
             return queryset
 
         filter_instance = self._get_filter_instance(filter_class, queryset, view)
-        rql_ast, queryset = filter_instance.apply_filters(
-            self.get_query(filter_instance, request, view), request, view,
-        )
+        query = self.get_query(filter_instance, request, view)
+
+        def apply_filters_lazy():
+            return filter_instance.apply_filters(query, request, view)
+
+        if filter_class.QUERIES_CACHE_BACKEND and request.method in ('GET', 'HEAD', 'OPTIONS'):
+            # We must use the combination of queryset and query to make a cache key as
+            #  queryset can already contain some filters (e.x. based on authentication)
+            cache_key = hash(str(queryset.query) + query)
```

Comment:
[SOL-C] Can we use not hashed  `str(queryset.query) + query` as a cache key? The possibility of hash collision is infinitely small, but still exists.